moor/lib/src/runtime/api/connection_user.dart

```diff
@@ -5,6 +5,8 @@ const _zoneRootUserKey = #DatabaseConnectionUser;
 typedef _CustomWriter<T> = Future<T> Function(
     QueryExecutor e, String sql, List<dynamic> vars);
 
+typedef _BatchRunner = dynamic Function(Batch batch);
```

Comment:
Thanks for your contribution :+1: 

Since we're not using the result of the function, let's also change the return type to void:

```suggestion
typedef _BatchRunner = void Function(Batch batch);
```

Reply:
ah cool, shall do!

Reply:
ah now I remember why I left it as `dynamic`, [the `if` statement](https://github.com/simolus3/moor/blob/519e8328c8f6170a6683bd66b5374f9ecafff219/moor/lib/src/runtime/api/connection_user.dart#L451) here checks the result of `batch`.

Is it fine to remove this check?

Reply:
What about returning FutureOr<void> and always do an await instead of the if statement?

Reply:
Ah true, I've missed that. Making it return `FutureOr<void>` should work, with or without the changes to the if statement.

Reply:
Thanks guys, change has been made.